android/app/src/main/java/org/fossasia/openevent/fragments/AboutFragment.java

```diff
@@ -318,6 +358,23 @@ public void onResume() {
         super.onResume();
         loadData();
     }
+    @Override
+    public void onAttach(Activity activity) {
+        super.onAttach(activity);
+
+        try {
+            mCallback = (OnMapSelectedListener) activity;
+        } catch (ClassCastException e) {
+            throw new ClassCastException(activity.toString()
+                    + "must implement OnMapSelectedListener");
+        }
```

Comment:
This isn't correct, the activity should explicitly pass its instance to the fragment through a method or `newInstance` static creator with concrete type of `OnMapSelectedListener`

This way we are not casting stuff in a statically compiled language.

Fragment should also check the nullability of the listener instance before calling it

Reply:
@iamareebjamal Can you please explain how to achieve the first part?
I am not able to understand 
And I get a null pointer exception when I check the nullability of the listener in the try block.
Other than this I have made all the other changes that you asked
Thank you

Reply:
This `onAttach` block should be removed. The activity should pass `this` i.e. the instance of `OnMapSelectedListener` to the Fragment when it is creating the fragment. So instead of `new Fragment()`, use a static method in fragment `newInstance(OnMapSelectedListener)` which will be called by the activity and will set the mapSelectedListener in the fragment

Wherever there'll be a call on listener, you'll have to check if it is null first